crates/goose/src/agents/subagent_execution_tool/task_execution_tracker.rs

```diff
@@ -93,6 +95,41 @@ impl TaskExecutionTracker {
             last_refresh: Arc::new(RwLock::new(Instant::now())),
             notifier,
             display_mode,
+            cancellation_token,
+        }
+    }
+
+    fn is_cancelled(&self) -> bool {
+        self.cancellation_token
+            .as_ref()
+            .is_some_and(|t| t.is_cancelled())
+    }
```

Comment:
we should probably put this in some util function and use everywhere but we can do that in something separate